Client/Frontend/Browser/Onboarding/OnboardingAdsCountdownViewController.swift

```diff
@@ -0,0 +1,87 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import BraveRewards
+import BraveRewardsUI
+import BraveShared
+
+class OnboardingAdsCountdownViewController: OnboardingViewController, UNUserNotificationCenterDelegate {
+    
+    private struct UX {
+        static let animationTime = 3.0
+    }
+    
+    private var timeSinceAnimationStarted: Date?
+    
+    private var contentView: View {
+        return view as! View // swiftlint:disable:this force_cast
+    }
+    
+    override func loadView() {
+        view = View(theme: theme)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        //Countdown timer
+        contentView.countdownText = String(UX.animationTime)
+        contentView.finishedButton.addTarget(self, action: #selector(continueTapped), for: .touchUpInside)
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        if let time = timeSinceAnimationStarted, Date().timeIntervalSince(time) >= UX.animationTime {
+            self.contentView.setState(.finished)
+            return
+        }
+        
+        timeSinceAnimationStarted = Date()
+        contentView.resetAnimation()
+        contentView.animate(from: 0.0, to: 1.0, duration: UX.animationTime) { [weak self] in
+            guard let self = self else { return }
+            self.displayMyFirstAdIfAvailable { action in
+                if action == .timedOut {
+                    //User possibly missed the ad.. show them confirmation screen.
+                    self.contentView.setState(.finished)
+                } else {
+                    //User saw the ad.. they interacted with it.. onboarding is finished.
+                    self.skipTapped()
+                }
+            }
+        }
+    }
+    
+    override func applyTheme(_ theme: Theme) {
+        styleChildren(theme: theme)
+        contentView.applyTheme(theme)
+    }
+}
+
+extension OnboardingAdsCountdownViewController {
+    
+    private func displayMyFirstAdIfAvailable(_ completion: ((AdsNotificationHandler.Action) -> Void)? = nil) {
+        if Preferences.Rewards.myFirstAdShown.value { return }
+
+        if BraveAds.isSupportedRegion(Locale.current.identifier) {
+            DispatchQueue.main.asyncAfter(deadline: .now() + 3) {
```

Comment:
This 3 second delay is not needed, since you already wait until the animation completes which is 3 seconds long

Reply:
@Brandon-T You marked this resolved but I don't see the change… unresolving for now

Reply:
Hmm for some reason it didn't go through when I committed via Xcode. Should be there this time.